### PR TITLE
[*] Mongoose - Handle last versions of mongoose by fixing the RangeOverflow error

### PR DESCRIPTION
**Pull request by ArnaudValensi**
Fix issue #113 

## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request
- [ ] Write changes made in the CHANGELOG.md
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

File: test/tests/adapters/expected-deep-nested.json [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            }
+          ]
+        }
+      },
+      {
+        "field": "_id",
+        "type": "String"
+      }
+    ]
+  }
```
Comment (arnaudbesnier) [inline test/tests/adapters/expected-deep-nested.json]: Strange Indentation, and missing newline at the end of the file.

File: adapters/mongoose.js [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
+            result[subKey] = keys.length - 1 === index ? data[key] : {};
+          } else {
+            result[subKey] = [];
+          }
+        }
+        return result[subKey];
+      }, result);
+    }
+    return result;
+  }
```
Comment (arnaudbesnier) [inline adapters/mongoose.js]: If this change fix other problems (as you mentioned in the Trello Card):
- it should be mentioned in the `CHANGELOG`
- a test about this other fix is more than welcome to:
  1. Explicit which case is fixed
  2. Prevent further regressions

File: test/tests/adapters/mongoose.js [hunk 3]
```diff
@@ -570,6 +576,52 @@ describe('Adapters > SchemaAdapter', function () {
...
+                      field1: [Date],
+                      field2: { field2Field1: Boolean },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      });
```
Comment (arnaudbesnier) [inline test/tests/adapters/mongoose.js]: Why didn't you also test with "nested schema" like we have in the model I sent?

**Comment by ArnaudValensi** [comment]
[P1 - RangeOverflow Error on forest-express-mongoose](https://trello.com/c/dM5cQaSV/2928-p1-rangeoverflow-error-on-forest-express-mongoose)

**Comment by arnaudbesnier** [comment]
@ArnaudValensi you can merge whenever you want.

**Pull request merged**